File: Fundamentos da Programação Orientada a Objetos/ArrayList_OrientadoAObjetos/ex02/entidades/Livraria.java
package entidades;

import java.util.ArrayList;

public class Livraria {
	 private ArrayList<Livro> livros;

	    public Livraria() {
	        livros = new ArrayList<>();
	    }

	    public void adicionarLivro(Livro livro) {
	        livros.add(livro);
	    }

	    public void removerLivro(Livro livro) {
	        livros.remove(livro);
	    }

	    public void renomearLivro(Livro livro, String novoNome) {
	        livro.setNome(novoNome);
	    }

	    public void exibirLivros() {
	        for (Livro livro : livros) {
	            livro.exibirLivro();
	            System.out.println("\n--------------------");
	        }
	    }
}

File: Programação Web Front-End/Lista_Array_POO/exerc_03/js/Livraria.js
export default class Livraria {
    #livros;

    constructor() {
        this.#livros = [];
    }

    adicionarLivro(livro) {
        this.#livros.push(livro);
    }

   
   removerLivro(codigo) {
    // Utilizamos o método filter() para criar um novo array 
    //que exclui o livro com o código especificado
    this.#livros = this.#livros.filter(function(livro){
        return (livro.getCodigo !== codigo);
    }); 
    // Exibimos uma mensagem informando que o livro foi removido
    console.log(`Livro com código ${codigo} removido.`);
}

listarLivros() {
    console.log("Livros disponíveis na livraria:");
    
    // Iteramos sobre o array #livros e exibimos informações sobre cada livro
    this.#livros.forEach(function(livro, indice) {
        console.log(`Livro ${indice + 1}:`);
        livro.exibirLivro();
    });
}
}